core/src/client/importer.rs

```diff
@@ -206,6 +207,8 @@ impl Importer {
         client.db().write_buffered(batch);
         chain.commit();
 
+        self.check_term_end(block.header(), &chain, client);
```

Comment:
This could make a problem.
In Tendermint Consensus, there will be proposal blocks which are executed but not finalized. Since the proposal blocks' timestamp will be different, some of it cause term end, and others do not cause term end. So the code, saving the "last term" to DB should not be called here. The code should be called when a block is finalized. And the finalization time would be "chain_notify".

@sgkim126 is handling a similar problem while implementing changing common params. Please check his work.

Reply:
Yes, it can be a problem when the block is a proposal block or there is a reorganization.
https://github.com/CodeChain-io/codechain/blob/master/core/src/client/client.rs#L150
To handle it, I extracted `new_blocks` method on the client.
I'm considering this method to check the params.

Reply:
However, you need to change `Importer::import_verified_blocks` to stop when the current block is the end of a term.

Reply:
I think I should change how the solo closes term. I think the solo should closes term still controllable manner (not non-deterministic time-based manner) but it is easily determined by just looking the header. Solo should closes term by block number base. The current manual trigger causes some problems. When the solo is configured with multiple nodes, they cannot agree when the term ends. and the differences between solo and the corgi make the problem harder such as how we deal with reorganization, when do we stop at on `import_verified_blocks`.